pink/rondis/string/db_operations.cc

```diff
@@ -614,3 +614,164 @@ int rondb_get_rondb_key(const NdbDictionary::Table *tab,
+    const NdbDictionary::Column *tot_value_len_col = tab->getColumn("tot_value_len");
+
+    NdbOperation::OperationOptions opts;
+    std::memset(&opts, 0, sizeof(opts));
+    /**
+     * The mask specifies which columns is to be updated after the interpreter
+     * has finished. The values are set in the key_row.
+     * We have 7 columns, we will update tot_value_len in interpreter, same with
+     * value_start.
+     *
+     * The rest, redis_key, rondb_key, value_data_type, num_rows and expiry_date
+     * are updated through final update.
+     */
+
+    const Uint32 mask = 0x57;
+    const unsigned char *mask_ptr = (const unsigned char *)&mask;
+
+    // redis_key already set as this is the Primary key
+    key_row->null_bits = 1; // Set rondb_key to NULL, first NULL column
+    key_row->num_rows = 0;
+    key_row->value_data_type = 0;
+    key_row->expiry_date = 0;
+
+    /* Define the interpreted program */
+    Uint32 code_buffer[128];
+    NdbInterpretedCode code(tab, &code_buffer[0], sizeof(code_buffer));
+    code.load_const_u16(REG0, 4); //Memory offset 0
+    code.load_const_u16(REG6, 0); //Memory offset 0
+    int ret_code = code.load_op_type(REG1); // Read operation type into register 1
+    code.branch_eq_const(REG1, RONDB_INSERT, LABEL0); //Inserts go to label 1
+
+    /**
+     * The first 4 bytes of the memory must be kept for the Attribute header
+     * REG0 Memory offset == 4
+     * REG1 Memory offset == 6
+     * REG2 Size of value_start
+     * REG3 Size of value_start without length bytes
+     * REG4 Old integer value after conversion
+     * REG5 New integer value after increment
+     * REG6 Memory offset == 0
+     * REG7 not used
+     */
+    /* UPDATE code */
+    code.read_full(value_start_col, REG6, REG2); // Read value_start column
+    code.load_const_u16(REG1, 6);//Memory offset 2
+    code.sub_const_reg(REG3, REG2, 2);//Subtract 2 from length
+    code.str_to_int64(REG4, REG1, REG3);//Convert string to number into register 6
+    code.add_const_reg(REG5, REG4, 1); //New integer value in register 6
+    code.int64_to_str(REG3, REG1, REG5);//Convert to string
+    code.add_const_reg(REG2, REG3, 2); //New value_start length
+    code.convert_size(REG3, REG0); //Write back length bytes in memory
+
+    code.write_interpreter_output(REG5, 0); //Write into output index 0
+    code.write_from_mem(value_start_col, REG6, REG2); // Write to column
+    code.write_attr(tot_value_len_col, REG3);
+    code.interpret_exit_ok();
+
+    /* INSERT code */
+    code.def_label(LABEL0);
+    code.load_const_u16(REG5, 1);
+    code.load_const_u16(REG3, 1);
+    code.write_interpreter_output(REG5, 0); //Write into output index 0
+
+    Uint32 insert_value;
+    Uint8 *insert_value_ptr = (Uint8*)&insert_value;
+    insert_value_ptr[0] = 1; // Length is 1
+    insert_value_ptr[1] = 0; // Second length byte is 0
+    insert_value_ptr[2] = '1'; //Inserts a string '1'
+    insert_value_ptr[3] = 0;
```

Comment:
Or is this the 1 we will add to the value?

Reply:
This is the bytes we will write into value_start, the string '1' with 2 length bytes representing a length of 1.

Reply:
This is used in an INSERT